work-tracker-spring/src/main/java/com/deere/isg/worktracker/spring/ServletEndpointRegistry.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright 2019 Deere & Company
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.deere.isg.worktracker.spring;
+
+import javax.servlet.ServletContext;
+import javax.servlet.ServletRegistration;
+import java.util.concurrent.ConcurrentHashMap;
+
+class ServletEndpointRegistry {
+    private static EndpointTrie trie = new EndpointTrie();
+
+    public static void populate(ServletContext context) {
+        if (context == null){
+            return;
+        }
+
+        context.getServletRegistrations()
+                .forEach((servletName, registration) -> addMappings(registration));
+    }
+
+    public static boolean contains(String value){
+        return trie.isEqual(value);
+    }
+
+    static void populate(String value){
+        trie.add(value);
+    }
+
+    static void clear(){
+        trie = new EndpointTrie();
+    }
+
+    private static void addMappings(ServletRegistration registration) {
+        registration.getMappings().forEach(ServletEndpointRegistry::populate);
+    }
+
+    private static class EndpointTrie{
```

Comment:
Just extracting this class to a util package somewhere would at least make it _seem_ less complex.

Reply:
Refactor this into Set